lib/graphql/schema/field.rb

```diff
@@ -666,7 +666,12 @@ def resolve(object, args, query_ctx)
               inner_object = obj.object
 
               if defined?(@hash_key)
-                inner_object[@hash_key] || inner_object[@hash_key_str] || (@fallback_value != :not_given ? @fallback_value : nil)
+                hash_value = inner_object.to_h.key?(@hash_key) ? inner_object[@hash_key] : inner_object[@hash_key_str]
+                if type.unwrap.graphql_name == "Boolean" && (hash_value == true || hash_value == false)
```

Comment:
Return `hash_value` in case it's a boolean field. `if type.unwrap.graphql_name == "Boolean" && hash_value == false` might be sufficient here!?

Reply:
Honestly, I don't think we even have to require that the field is boolean. Something like this would be ok with me too, I think: 

```suggestion
                if hash_value == false
```

Because maybe the application has some use for a `false` value _other_ than the built-in `Boolean` type. What do you think?

Reply:
Agree. Didn't think about that case.